### PR TITLE
Add storageClient API definitions for TozID EACP 

### PR DESCRIPTION
**Pull request by galxy25**
- Update storageClient API definitions for activate challenge endpoint to include more fully specified activated challenges.
- Fix identity api struct semantic error.

### CODE REVIEW

File: storageClient/storageClient_test.go [hunk 1]
```diff
@@ -49,9 +48,31 @@ func TestFullSignatureAuthenticationFlowSucceedsWithUID(t *testing.T) {
...
 	}
-	_, err = StorageClient.WriteNote(ctx, *noteToWrite)
+	_, err = StorageClient.WriteNote(testCtx, *noteToWrite)
 	if err != nil {
-		t.Errorf("unable to write note %+v, to storage servive %s\n", noteToWrite, err)
+		t.Errorf("unable to write note %+v, to storage service %s\n", noteToWrite, err)
+	}
+}
+
+func TestAddTozIDEACPWhenCreatingNote(t *testing.T) {
```
Comment (efabens) [inline storageClient/storageClient_test.go]: What's a Toz Idea cpw 🤣

File: storageClient/api.go [hunk 2]
```diff
@@ -80,9 +85,36 @@ type PrimeResponseBody struct {
...
+type ChallengeResponse struct {
+	EmailEACPChallenge *EmailEACPChallengeResponse `json:"email_eacp,omitempty"`
+	TozIDEACPChallenge *TozIDEACPChallengeResponse `json:"tozid_eacp,omitemtpy"`
+}
+
+type EmailEACPChallengeResponse struct {
+	EmailAddress string `json:"email_address"`
+	TemplateName string `json:"template"`
+	ProviderLink string `json:"provider_link"`
+	OTP          string `json:"otp"`
```
Comment (lkwdwrd) [inline storageClient/api.go]: We definitely don't want to return the OTP or provider link from this EACP -- it should only be delivered via the email channel out of band of this request flow.

Comment (lkwdwrd) [reply storageClient/api.go]: As mentioned in the API spec, I'm actually thinking we don't want to respond with any of this information. Happy to discuss, but in general this should be kept to a very minimum.

File: storageClient/api.go [hunk 3]
```diff
@@ -80,9 +85,36 @@ type PrimeResponseBody struct {
...
+}
+
+// TozIDEACPChallengeRequest wraps parameters for an activated TozID EACP on a note.
+type TozIDEACPChallengeResponse struct {
+	ExpiresAt time.Time `json:"expires_at"`
+	OTP       string    `json:"otp"`
```
Comment (lkwdwrd) [inline storageClient/api.go]: change name to something indicating this is a challenge value that must be signed over in an identity token as the nonce value.

**Pull request merged**